src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -698,6 +713,61 @@ namespace cryptonote
 
     return true;
   }
+
+  tx_extra_service_node_deregister_ convert_legacy_tx_extra_deregister(tx_extra_service_node_deregister_legacy const &legacy)
```

Comment:
A `explicit` converting constructor here (or a converting `explicit operator tx_extra_service_node_deregister_()` in the legacy one) would be nicer, then you can just do: `auto newreg = static_cast<tx_extra_service_node_deregister_>(oldreg);`